### PR TITLE
Update disability verification dates based on QaaWS

### PR DESCRIPTION
**Pull request by eanders**

### CODE REVIEW

File: app/models/bo/client_id_lookup.rb [hunk 1]
```diff
@@ -265,6 +266,69 @@ def call_with_retry settings, message_options
...
+      return unless @config.disability_verification_cuid.present?
+      @disability_verifications = fetch_disability_verifications.
+        group_by{ |row| row[:participant_enterprise_identifier].gsub('-', '') }
+      personal_ids = @disability_verifications.keys
+      source_clients = GrdaWarehouse::Hud::Client.source.where(
+        data_source_id: @data_source_id,
+        PersonalID: personal_ids
+        ).select(:id, :PersonalID, :disability_verified_on)
+      updated_source = 0
+      updated_destination = 0
```
Comment (awisema) [inline app/models/bo/client_id_lookup.rb]: I would prefer these names to make it clear they are counts

File: app/models/bo/client_id_lookup.rb [hunk 2]
```diff
@@ -265,6 +266,69 @@ def call_with_retry settings, message_options
...
+        # only set the verification date if it was blank before or is newer
+        # then, check the destination client and update that as well
+        # We could batch this to improve performance if necessary, but after the first load
+        # this should only be a handful of clients each day
+        if client.disability_verified_on.blank? || max_date > client.disability_verified_on
+          client.update(disability_verified_on: max_date)
+          updated_source += 1
+          dest_client = client.destination_client
+          # if the source client changed, reflect changes on the destination client
+          if dest_client.disability_verified_on.blank? || client.disability_verified_on > dest_client.disability_verified_on
```
Comment (awisema) [inline app/models/bo/client_id_lookup.rb]: The code and comment don't quite match up... Maybe something like
`# reflect changes on the destination client if the source client data is newer`

File: app/models/bo/client_id_lookup.rb [hunk 3]
```diff
@@ -265,6 +266,69 @@ def call_with_retry settings, message_options
...
+            dest_client.update(disability_verified_on: max_date)
```
Comment (awisema) [inline app/models/bo/client_id_lookup.rb]: I think if the test is for `client.disability_verified_on > ...`, the update should use that as well...

File: app/models/bo/client_id_lookup.rb [hunk 4]
```diff
@@ -265,6 +266,69 @@ def call_with_retry settings, message_options
...
+      @notifier.ping msg if send_notifications && msg.present?
+      settings = {
+        url: "#{@config.url}?wsdl=1&cuid=#{@config.disability_verification_cuid}",
+        method: :disability_lookup,
+      }
+      message_options = {
+        touch_point_id: @config.disability_touch_point_id,
+        touch_point_question_id: @config.disability_touch_point_question_id,
+      }
+      if one_off
```
Comment (awisema) [inline app/models/bo/client_id_lookup.rb]: Tempting to suggest calling `call_with_retry` here with a new parameter `retries` - it just looks odd to build up the `settings` and `message_options` objects, and then tear them apart again.

File: app/models/bo/soap.rb [hunk 5]
```diff
@@ -20,6 +20,13 @@ def site_touch_point_lookup url, options: {}
       data_from url, site_touch_point_lookup_xml
     end
 
+    def disability_lookup url, touch_point_id:, touch_point_question_id:
+      data_from url, disability_lookup_xml(
```
Comment (awisema) [inline app/models/bo/soap.rb]: Would prefer `data_from(url, ...` I found this hard to read

File: app/models/bo/client_id_lookup.rb (hunk 4 above)
Comment (eanders) [reply app/models/bo/client_id_lookup.rb]: re-written with a `call_once` method to use the same structure as `call_with_retry` but not do the retries.

File: app/models/bo/client_id_lookup.rb (hunk 1 above)
Comment (eanders) [reply app/models/bo/client_id_lookup.rb]: done

**Pull request merged**